x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -229,7 +229,7 @@ export const AlertsTableComponent: FC<DetectionEngineAlertTableProps> = ({
     [alertColumns, isEventRenderedView]
   );
 
-  const onAlertTableUpdate: AlertsTableStateProps['onUpdate'] = useCallback(
+  const onAlertTableUpdate = useCallback<NonNullable<AlertsTableStateProps['onUpdate']>>(
```

Comment:
May I ask why `NonNullable` is needed here?

Reply:
Sure! When assigned like this `const onAlertTableUpdate: AlertsTableStateProps['onUpdate']` 
the argument in the function is `any` 

to fix it, we assign like this: `const onAlertTableUpdate = useCallback<AlertsTableStateProps['onUpdate']>`

But this gives an error because AlertsTableStateProps['onUpdate'] could be undefined and it can't be used as a generic param. So with NonNullable we exclude undefined from that type

Reply:
Perfect. Thanks for explanation. I understand now.